File: Models/Payment.cs
using Newtonsoft.Json;
using System;

namespace KajfestPOS.Models
{
    public class Payment
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        [JsonIgnore]
        public Order Order { get; set; }
        public PaymentMethod Method { get; set; }
        public decimal Amount { get; set; }
        public int? AccountId { get; set; }
        [JsonIgnore]
        public Account Account { get; set; }
        public DateTimeOffset Created { get; set; }
    }

    public enum PaymentMethod
    {
        Card = 1,
        Cash = 2,
        Account = 3
    }
}

File: src/FestivalPOS/src/api/payment.ts
import { Big } from "big.js";
import { DateTime } from "luxon";
import { bigConverter } from "./converters/big-converter";
import { dateTimeConverter } from "./converters/date-time-converter";
import { jsonProperty } from "@utiliread/json";

export type PaymentMethod = "card" | "cash" | "account" | "mobilePay";

export class Payment {
    @jsonProperty()
    id!: number;

    @jsonProperty()
    orderId?: number;

    @jsonProperty()
    method!: PaymentMethod;

    @jsonProperty({ converter: bigConverter })
    amount!: Big;

    @jsonProperty()
    transactionNumber?: string;

    @jsonProperty()
    accountId?: number;

    @jsonProperty({ converter: dateTimeConverter })
    created!: DateTime;
}